File: app/src/main/java/com/ybj/myapplication/java/arithmetic/Lesson1_3.java
package com.ybj.myapplication.java.arithmetic;

import com.ybj.myapplication.java.ArithmeticTest;

/**
 * Created by 杨阳洋 on 2020/7/7.
 */
public class Lesson1_3 implements ArithmeticTest {


    public static void main(String [] args){
        Lesson1_3 lesson1_1 = new Lesson1_3();
        lesson1_1.test();
    }

    @Override
    public void test() {
        int a[] = {1,2,3,4,5,4,3,2,1,2};
        int time_max = 0;
        int val_max = -1;
        int time_tmp = 0;
        for (int i = 0 ; i < a.length ; i++){
            time_tmp = 0;
            for (int j = 0 ; j < a.length ; j++){
                if(a[i] == a[j]) {
                    time_tmp += 1;
                }
            }
            if(time_tmp > time_max) {
                time_max = time_tmp;
                val_max = a[i];
            }

        }
        System.out.println(val_max);

//        HashMap<Integer,Integer> hashMap = new HashMap();
//        for (int i = 0; i < a.length ; i++){
//            if(hashMap.containsKey(a[i])) {
//                hashMap.put(a[i],hashMap.get(a[i])+1);
//            }else{
//                hashMap.put(a[i],1);
//            }
//        }
//
//        for (Integer key: hashMap.keySet()){
//            if(hashMap.get(key) > time_max) {
//                time_max = hashMap.get(key);
//                val_max = key;
//            }
//        }
//
//        System.out.println(val_max);
    }

}

File: app/src/main/java/com/ybj/myapplication/kotlin/arithemtic/Lesson1_3.kt
package com.ybj.myapplication.kotlin.arithemtic

/**
 * Created by 杨阳洋 on 2020/7/7.
 * 查找数组中最大元素
 */
class Lesson1_3 {
    fun test() {
        val a = intArrayOf(1, 2, 3, 4, 5, 4, 3, 2, 1, 2)
        var timeMax = 0
        var valMax = -1
        var timeTmp = 0
        for (i in a.indices) {
            timeTmp = 0
            for (j in a.indices) {
                if (a[i] == a[j]) {
                    timeTmp += 1
                }
            }
            if (timeTmp > timeMax) {
                timeMax = timeTmp
                valMax = a[i]
            }
        }
        println(valMax)
    }
}
